server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationState.java

```diff
@@ -556,27 +556,18 @@ public interface PersistedState extends Closeable {
          * marked as committed.
          */
         default void markLastAcceptedStateAsCommitted() {
-            final ClusterState lastAcceptedState = getLastAcceptedState();
-            Metadata.Builder metadataBuilder = null;
-            if (lastAcceptedState.getLastAcceptedConfiguration().equals(lastAcceptedState.getLastCommittedConfiguration()) == false) {
-                final CoordinationMetadata coordinationMetadata = CoordinationMetadata.builder(lastAcceptedState.coordinationMetadata())
-                    .lastCommittedConfiguration(lastAcceptedState.getLastAcceptedConfiguration())
-                    .build();
-                metadataBuilder = Metadata.builder(lastAcceptedState.metadata());
-                metadataBuilder.coordinationMetadata(coordinationMetadata);
-            }
-            assert lastAcceptedState.metadata().clusterUUID().equals(Metadata.UNKNOWN_CLUSTER_UUID) == false
-                : "received cluster state with empty cluster uuid: " + lastAcceptedState;
-            if (lastAcceptedState.metadata().clusterUUID().equals(Metadata.UNKNOWN_CLUSTER_UUID) == false
-                && lastAcceptedState.metadata().clusterUUIDCommitted() == false) {
-                if (metadataBuilder == null) {
-                    metadataBuilder = Metadata.builder(lastAcceptedState.metadata());
-                }
-                metadataBuilder.clusterUUIDCommitted(true);
+            final var lastAcceptedState = getLastAcceptedState();
+            final var hasClusterUuid = lastAcceptedState.metadata().clusterUUID().equals(Metadata.UNKNOWN_CLUSTER_UUID) == false;
+            assert hasClusterUuid : "received cluster state with empty cluster uuid: " + lastAcceptedState;
+
+            if (hasClusterUuid && lastAcceptedState.metadata().clusterUUIDCommitted() == false) {
```

Comment:
Feels odd to protect the logging by something we assert on above?

```suggestion
            if (lastAcceptedState.metadata().clusterUUIDCommitted() == false) {
```

Reply:
Ehh I just kept the same logic as before but I don't think I've ever seen this assertion trip. I'd be ok to drop it.